lib/WebDriver/AbstractWebDriver.php

```diff
@@ -128,38 +128,44 @@ protected function curl($requestMethod, $command, $parameters = null, $extraOpti
             throw WebDriverException::factory(WebDriverException::CURL_EXEC, 'Webdriver http error: ' . $info['http_code'] . ', payload :' . substr($rawResult, 0, 1000));
         }
 
-        $result = json_decode($rawResult, true);
+        $result = [];
+        $value = null;
+        if (!empty($rawResult)) {
+            $result = json_decode($rawResult, true);
 
-        if ($result === null && json_last_error() != JSON_ERROR_NONE) {
-            throw WebDriverException::factory(WebDriverException::CURL_EXEC, 'Payload received from webdriver is not valid json: ' . substr($rawResult, 0, 1000));
-        }
+            if ($result === null && json_last_error() != JSON_ERROR_NONE) {
```

Comment:
I'm going to rewrite this since we want to push a fix tonight while maintaining our coding standards (eg no nested if).